### PR TITLE
docs: streaming documentation

### PR DESCRIPTION
**Pull request by alaeddine-13**
fixes for streaming documentation and more elaborated examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -342,6 +341,120 @@ response[0].display()
 
 <!-- end build-pipelines -->
 
+### Streaming for LLMs
+<!-- start llm-streaming-intro -->
+Large Language Models can power a wide range of applications from chatbots to assistants and intelligent systems.
+However, these models can be heavy and slow and you want your users to experience fast yet intelligent systems!
+
+A large language models works by turning your question into tokens and then will keep generating the next token each 
+time until it believes the generation should stop.
```
Comment (scott-martens) [inline README.md:350]:
```suggestion
Large language models work by turning your questions into tokens and then generating new token one at a 
time until it decides that generation should stop.
```

File: README.md [hunk 2]
```diff
@@ -342,6 +341,120 @@ response[0].display()
...
+        super().__init__(**kwargs)
+        self.model = GPT2LMHeadModel.from_pretrained('gpt2')
+```
+<!-- end llm-streaming-init -->
+
+
+#### Implement the streaming endpoint
+<!-- start llm-streaming-endpoint -->
+Our streaming endpoint accepts a `PromptDocument` as input and streams `TokenDocument`s. To stream a document back to 
+the client, you can use the `yield` keyword in the endpoint implementation. Therefore, we use the model to generate
```
Comment (scott-martens) [inline README.md:397]:
```suggestion
the client, use the `yield` keyword in the endpoint implementation. Therefore, we use the model to generate 
```

File: README.md [hunk 3]
```diff
@@ -342,6 +341,120 @@ response[0].display()
...
+        self.model = GPT2LMHeadModel.from_pretrained('gpt2')
+```
+<!-- end llm-streaming-init -->
+
+
+#### Implement the streaming endpoint
+<!-- start llm-streaming-endpoint -->
+Our streaming endpoint accepts a `PromptDocument` as input and streams `TokenDocument`s. To stream a document back to 
+the client, you can use the `yield` keyword in the endpoint implementation. Therefore, we use the model to generate 
+`max_tokens` tokens and yield them until the generation stops:
```
Comment (scott-martens) [inline README.md:398]:
```suggestion
up to `max_tokens` tokens and yield them until the generation stops: 
```

File: README.md [hunk 4]
```diff
@@ -342,6 +341,120 @@ response[0].display()
...
+        if output[0][-1] == tokenizer.eos_token_id:
+            break
+        yield TokenDocument(token_id=output[0][-1])
+        encoded_input = {
+            'input_ids': output,
+            'attention_mask': torch.ones(1, len(output[0])),
+        }
+```
+
+Learn more about {ref}`streaming endpoints <streaming-endpoints>`.
```
Comment (scott-martens) [inline README.md:414]:
```suggestion
Learn more about {ref}`streaming endpoints <streaming-endpoints>` from the `Executor` documentation.
```

File: README.md [hunk 5]
```diff
@@ -342,6 +341,120 @@ response[0].display()
...
+<!-- end llm-streaming-endpoint -->
+
+
+#### Serve and send requests
```
Comment (scott-martens) [inline README.md:418]:
```suggestion
#### Send and serve requests
```

File: docs/tutorials/llm-serve.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+## Service Schemas
+```{include} ../../README.md
+:start-after: <!-- start llm-streaming-schemas -->
+:end-before: <!-- end llm-streaming-schemas -->
+```
+
+```{admonition} Note
+:class: note
+
+Thanks to DocArray's flexibility, you can actually implement much more flexible services. For instance, you can use
```
Comment (scott-martens) [inline docs/tutorials/llm-serve.md:16]:
```suggestion
Thanks to DocArray's flexibility, you can implement very flexible services. For instance, you can use 
```

File: README.md (hunk 5 above)
Comment (alaeddine-13) [reply README.md:418]: actually we serve the executor first than we send requests to it

Comment (scott-martens) [reply README.md:418]: It sounds better in the other order in English, but it's not a big deal either way. Leave it as you like.

File: README.md [hunk 7]
```diff
@@ -342,6 +341,120 @@ response[0].display()
 
 <!-- end build-pipelines -->
 
+### Streaming for LLMs
+<!-- start llm-streaming-intro -->
+Large Language Models can power a wide range of applications from chatbots to assistants and intelligent systems.
+However, these models can be heavy and slow and your users want systems that are both intelligent _and_ fast!!
```
Comment (scott-martens) [inline README.md:347]:
```suggestion
However, these models can be heavy and slow and your users want systems that are both intelligent _and_ fast!
```

File: README.md [hunk 8]
```diff
@@ -342,6 +341,120 @@ response[0].display()
...
+
+Large language models work by turning your questions into tokens and then generating new token one at a 
+time until it decides that generation should stop.
+This means you want to **stream** the output tokens generated by a large language model to the client. 
+In this tutorial, we will discuss how to achieve this with Streaming Endpoints in Jina!
```
Comment (scott-martens) [inline README.md:352]:
```suggestion
In this tutorial, we will discuss how to achieve this with Streaming Endpoints in Jina.
```

**Review by scott-martens — APPROVED** [review]
LGTM

**Review by JoanFM — APPROVED** [review]

**Pull request merged**